### ISSUE TITLE
VERSION grepping seems broken with new build 42 game versioning(duplicate versioning and newline character)

### ISSUE THREAD

**Issue by sam-trafton-dev** [quarantin/beautiful-java#2]
## Reproduce
Using Windows, JDK 17, and Build 42.0 of Project Zomboid. With this environment set up running `BeautifulJava` result in the errors during decompile process:

- missing: 'C:/${PATH}/beautiful-java/pz/42.0 42.0/classes', ignored
- Error: no sources given
- cp: cannot stat 'C:/${PATH}/beautiful-java/pz/**42.0'$'\n''42.0**/decompiled/*': No such file or directory

## Hacky Fix
I don't know much about bash or shell scripting so I just trimmed the newline character that is generating in the directory path and made sure no spaces.
```
VERSION="$(grep -o 'new GameVersion(.*' "${TMPDIR}/Core.java" | cut -f2 -d'(' | cut -f1,2 -d',' | sed 's/, /./' | tr -d '\n' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//')"

if [ -z "${VERSION}" ]; then
    VERSION="$(grep 'versionNumber =' "${TMPDIR}/Core.java" | cut -f2 -d= | cut -f2 -d'"' | tr -d '\n' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//')"
fi
```

This allows the program to execute and decompile the files successfully but you end up with a directory titled 42.042.0 

# Full Fix
Not sure what conditional logic would need to be placed in but I'm assuming it has to do with the fact that build 42 in the Core.class file lists the build as: `private static final GameVersion gameVersion = new GameVersion(42, 0, "");` which maybe in the past had three characters that could be grepped? I also noticed 
```
VERSION_MAJOR="$(echo "${VERSION}" | cut -f1 -d.)"
VERSION_MINOR="$(echo "${VERSION}" | cut -f2 -d.)"
```
Are unused in `version.sh` but seems like the right direction for fixing by appending major and minor together before generating a directory name. Let me know if you'd like help resolving or want me to send a PR I'm pretty new to development but this is how I learn. Merry Christmas!

**Comment by quarantin** [comment]
Hi, thank you for your interest in BeautifulJava and sorry for answering late.
FYI I just pushed a modification that should fix the issue you describe.
The problem was that in previous versions of Project Zomboid, the version string was present only once in the source file. Now it's present twice as you can see:
```
$ grep -o 'new GameVersion(.*' pz/*/decompiled/zombie/core/Core.java
new GameVersion(42, 0, "");
new GameVersion(42, 0, "");
```
So now the updated command only keeps the first line.
On the other hand there's another issue preventing the renaming of symbols to something meaningful, but in the meantime you can still access the raw decompilation in `pz/42.0/decompiled/`.

Feel free to open another issue if I take too long to fix it :sweat_smile:

**Comment by quarantin** [comment]
@sam-trafton-dev FYI I pushed a new release to fix the renaming of symbols.
Now you should be able to access the source in `pz/42.0/sources/` :wink:
